core/state/history.go

```diff
@@ -0,0 +1,101 @@
+package state
+
+import (
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/trie2"
+	"github.com/NethermindEth/juno/db"
+)
+
+var _ StateReader = &StateHistory{}
```

Comment:
Is this to make sure it implements an interface?

```
var _ StateReader = (*StateHistory)(nil)
```

You should use this so it doesn't use any memory at runtime and the check is still made at compile time